src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.cs

```diff
@@ -173,17 +175,28 @@ internal bool NotifyWorkItemComplete()
             Interlocked.Increment(ref _completionCount);
             Volatile.Write(ref _separated.lastDequeueTime, Environment.TickCount);
 
-            bool shouldAdjustWorkers = ShouldAdjustMaxWorkersActive();
-            if(shouldAdjustWorkers)
+            bool acquiredLock = _hillClimbingThreadAdjustmentLock.TryAcquire();
+            try
             {
-                AdjustMaxWorkersActive();
+                bool shouldAdjustWorkers = ShouldAdjustMaxWorkersActive() && acquiredLock;
```

Comment:
ShouldAdjustMaxWorkersActive and ShouldStopProcessingWorkNow don't need to be inside the lock